packages/select/src/components/multi-select/multiSelect.tsx

```diff
@@ -334,14 +371,20 @@ export class MultiSelect<T> extends AbstractPureComponent<MultiSelectProps<T>, M
     // Note that we defer to the next animation frame in order to get the latest activeElement
     private handlePopoverInteraction = (nextOpenState: boolean, evt?: React.SyntheticEvent<HTMLElement>) =>
         this.requestAnimationFrame(() => {
-            const isInputFocused = this.input === Utils.getActiveElement(this.input);
-
-            if (this.input != null && !isInputFocused) {
-                // input is no longer focused, we should close the popover
-                this.setState({ isOpen: false });
-            } else if (!this.props.openOnKeyDown) {
-                // we should open immediately on click focus events
-                this.setState({ isOpen: true });
+            // Cannot rely on input to determine popover state as the input will be inside the Popover
+            // if customTarget is provided
+            if (this.props.customTarget != null) {
+                this.setState({ isOpen: nextOpenState });
```

Comment:
is this actually needed? it looks like we don't always set the state here currently. maybe we just early return in the `customTarget` case

Reply:
Yeah this is needed, we control the Popover state via the `isOpen` state.  This is the only place where thats updated

Currently we do update the state from click events here, just not sure why we don't directly use the `nextOpenState` rather than doing these checks for whether whether that TagInput is focused or not

Reply:
I see - it seems this `requestAnimationFrame` is for the input focus, which we don't need

maybe in this `requestAnimationFrame` we should be returning early if `customTarget` is defined and instead always directly setting `isOpen` outside of the `requestAnimationFrame` - it looks like that is what the `Select` component does

---

I'm not sure why this is checking input focus and not using the `nextOpenState` but I think it's safest to just let that code be

Reply:
sorry I'm realizing this may not have been clear but I was expecting to see this early return block outside of the `requestAnimationFrame` - have you tried that by any chance? wondering if that's why the tests need the `await`